File: 0686_repeated_string_match/main.go
package main

import (
	"math"
	"strings"
)

func repeatedStringMatch(A string, B string) int {
	times := int(math.Ceil(float64(len(B)) / float64(len(A))))
	match := strings.Repeat(A, times)
	if strings.Contains(match, B) {
		return times
	}
	match = match + A
	if strings.Contains(match, B) {
		return times + 1
	}
	return -1
}

func main() {

}

File: 0686_repeated_string_match/main.py
class Solution:
    def repeatedStringMatch(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: int
        """
        times = math.ceil(len(B) / len(A))
        match = A * times
        if B in match:
            return times
        match += A
        times += 1
        if B in match:
            return times
        return -1